### ISSUE TITLE
Round beats to 2 decimal places in screen reader speech

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#34401]
### Your idea

When navigating over a note in a tuplet, the screen reader currently might say:

> _Note G4 half, measure 1, beat 3.666667_

To make navigation more snappy for blind users, we could truncate the beat to 2 decimal places.

> _Note G4 half, measure 1, beat 3.67_

Alternatively, we could give it as a mixed fraction. such as one of these options:

> _Note G4 half, measure 1, beat 3 ²⁄₃_

> _Note G4 half, measure 1, beat three and two thirds_

> _Note G4 half, measure 1, beat 3 and 2 over 3_

### Problem to be solved

The status bar displays a beat number with many decimal places when the score contains:

* Triplets or certain other tuplets.

    <img width="518" height="210" alt="Image" src="https://github.com/user-attachments/assets/2167a048-eb76-42a3-938c-c87e73ee7f05" />

* Ordinary notes or rests of a 64th or smaller duration.

     <img width="546" height="196" alt="Image" src="https://github.com/user-attachments/assets/b0e8de42-38a2-4cbb-aa13-d3da3d357574" />

It's not really a problem for sighted users, who can look past those digits to get the information they need.

However, blind users have to listen to all of these digits as they move through the score, which makes navigation much slower than it needs to be.

**Note:** We can change what blind users listen to without changing what is shown in the status bar.

### Additional context

If truncating to 2 decimal places, beats with fewer decimal places shouldn't be expanded.

* A whole-number beat like 1 should be spoken as 1 and not 1.00.
* A single decimal like 1.5 should be spoken as 1.5 and not 1.50.

If using fractions, [Unicode trickery](https://onlineminitools.com/create-unicode-fractions?text=2+7%2F23) like 2 ⁷⁄₂₃ might result in bad speech with some screen readers:

> _Beat 2, superscript 7, fraction slash, subscript 2, subscript 3._

Instead, we'd probably have to give the screen reader the exact words we want it to say, such as:

> _Beat two and seven twenty-thirds._

Technically it's difficult to give a string for every possible fraction, and it would be difficult to translate into other languages besides English, so something like this would be much easer:

> _Beat 2 and 7 over 23._

### Checklist

- [x] This request follows the <a href="https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues" target="_blank">guidelines</a> for reporting issues
- [x] I have verified that this feature request has not been logged before, by searching the <a href="https://github.com/musescore/MuseScore/issues" target="_blank">issue tracker</a> for similar requests

**Comment by shoogle** [comment]
I think the fractional representation is probably best for screen readers, but we could add an option in **Preferences > Accessibility** to go back to using decimals (now rounded to 2dp).

We could also use fractions in the Status bar, but this would require testing the font on all platforms to ensure they render correctly.

**Comment by stevage** [comment]
>We could also use fractions in the Status bar, but this would require testing the font on all platforms to ensure they render correctly.

Decimals do have some advantages over fractions - it's easier to quickly tell which of two numbers is bigger, and to reason about. For instance, it's instantly obvious that a note that is at beat 0.667 comes slightly before a note at beat 0.75, requires a moment of thought to see that that 2/3 is less than 3/4. 

Also, for me I value the fact that the status bar representation of the beat is different from what I can read by looking at the notation. 

Personally, if there was an option for fractions or decimals in the status bar, I'd be choosing the decimals option.

**Comment by shoogle** [comment]
Decimal beat numbers are also less likely to be confused with musical durations (which we call "ticks" in the code, where 1 tick equals 1 whole note).

For example, beat 1.5 is obviously midway between beats 1 and 2.

But beat 1 ¹⁄₂ might be confused with "one half note" or "beat 1 plus a half note", when it's really just beat 1.5 expressed as a fraction.

**Comment by stevage** [comment]
> where 1 tick equals 1 whole note

I think you mean 1920 ticks equals 1 whole note.

**Comment by jmwmusic** [comment]
I think the decimal points are much easier to understand in speech than any of the fractions.  The over and divided by are both adding more words and not how many of us would ever speak those. "two thirds" is OK, but going to get very confusing when smaller triplets and tuplets such as semiquavers or demisemiquavers.  It also means processing two different numbers in your mind, the beat itself and separately the portion of the beat rather than all in one number.  The discussions in the Daisy team though were for 3 decimal places rather than 2, so down to demisemiquaver triplets included.
